### PR TITLE
use w and h parameters of panel Paint hooks where applicable

### PR DESCRIPTION
**Pull request by zerfgog**

### CODE REVIEW

File: garrysmod/gamemodes/terrortown/gamemode/vgui/sb_main.lua [hunk 1]
```diff
@@ -180,12 +180,12 @@ local colors = {
...
 
-function PANEL:Paint()
+function PANEL:Paint(w, h)
    -- Logo sticks out, so always offset bg
-   draw.RoundedBox( 8, 0, y_logo_off, self:GetWide(), self:GetTall() - y_logo_off, colors.bg)
+   draw.RoundedBox( 8, 0, y_logo_off, w, h - y_logo_off, colors.bg)
 
    -- Server name is outlined by orange/gold area
-   draw.RoundedBox( 8, 0, y_logo_off + 25, self:GetWide(), 32, colors.bar)
+   draw.RoundedBox( 8, 0, y_logo_off + 25, h, 32, colors.bar)
```
Comment (zerfgog) [inline garrysmod/gamemodes/terrortown/gamemode/vgui/sb_main.lua]: whoops

**Comment by Bo98** [comment]
Just checked - some addons do indeed draw Paint manually:

https://github.com/FPtje/DarkRP/blob/master/gamemode/modules/fadmin/fadmin/playeractions/giveweapons/cl_controls.lua#L45

(I will send a pull request for this particular instance but I can't fix all addons)

**Comment by Kefta** [comment]
Why not just say "if not w then w = self:GetWide()"? Same as default values for C++ methods.

**Comment by robotboy655** [comment]
Due to the problem Bo98 outlined, I will not merge this.

@Kefta Because it doesn't make any sense.

**Pull request closed without merging**